clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractHeartbeatRequestManager.java

```diff
@@ -379,7 +379,9 @@ private void onErrorResponse(final R response, final long currentTimeMs) {
             case GROUP_MAX_SIZE_REACHED:
             case UNSUPPORTED_ASSIGNOR:
             case UNSUPPORTED_VERSION:
-                logger.error("{} failed due to {}: {}", heartbeatRequestName(), error, errorMessage);
+                logger.error("{} failed due to {}: {}. The cluster doesn't yet support the new consumer group protocol." +
+                        " Set group.protocol=classic to revert to the classic protocol until the cluster is upgraded.",
+                        heartbeatRequestName(), error, errorMessage);
```

Comment:
This does not sound right to me because it prints the errors for the other errors too. We should only have it for `UNSUPPORTED_VERSION`.